Recipe App

ogagaoghene opened the pull request:
# Recipe App

## Description
In this PR, I implemented the following requirements.
- Login page and registration page:
- Should be built with Devise.
Food list:
- Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also --quantity of a given food).
- Should lead to a form that allows users to add new food.
Recipes list:
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.
Public recipe list:
- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.
Recipe details:
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
General shopping list view (only for a group that has less than 3 members):
- Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.

fobadara reviewed (commented):
Hi  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is not making use of the new README

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

tresorsawasawa reviewed (changes requested):
# STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @ ,:wave: :wave: 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 🛑

You have submitted a project that cannot be running. Below is ther error shown when trying to run migrations after creating a new database.
![Screenshot from 2022-11-25 01-30-57](https://user-images.githubusercontent.com/83042742/203874996-077c503f-7d93-4342-8e75-9cc11e520717.png)


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

---
Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

---
_Invalid Code Review Request **Does not count** into the code reviews limit._

ogagaoghene commented:
@tresorsawasawa 
Please don't run  migrations. It is not necessary. I just added that in the README..md as part of the requirement for SETUP. I have removed it. Please download and spin up the server. For some reason you might want to run "NPM install" because I am using Bootstrap CDN. 

Please find images of the working application on my local machine. 

![image](https://user-images.githubusercontent.com/30408871/203876003-5562eec3-14fc-461e-8165-51b5de202d1a.png)
![image](https://user-images.githubusercontent.com/30408871/203876026-b22e6c84-2cfa-447d-b989-1a25a82bb0ee.png)
![image](https://user-images.githubusercontent.com/30408871/203876052-1fa5601d-5e0b-416f-9284-dc7fbb8518c1.png)
![image](https://user-images.githubusercontent.com/30408871/203876068-bb4c482b-d8a0-4fbe-897f-4e7a5453ebda.png)
![image](https://user-images.githubusercontent.com/30408871/203876087-c7d20aaf-7240-408f-baa0-e50ffb1d7a2a.png)
![image](https://user-images.githubusercontent.com/30408871/203876100-a1de00ce-0ebd-4aae-8851-0fd729c1622c.png)
![image](https://user-images.githubusercontent.com/30408871/203876139-dc9a9dfe-042e-4944-abe2-190d4062be0a.png)

Kindly review my project as it is working locally. "Just run rails server, sign up, request for confirmation, open letter local smtp server launches asking for email confirmation. Log on with the confirmed credential. You can start populating or testing out the application. It is a working application. 

Thanks in anticipation for a positive review

ogagaoghene commented:
@tresorsawasawa Don't follow my README strictly. It might be different. You don't need to run db:migrate bro. It was a mistake on my part to have included it in my README file. I did not know you guys follow the README strictly to do your review. Please disregard the command "rails db:migrate". All migrations are up and it won;t allow fot that.

ogagaoghene commented:
@tresorsawasawa You can populate it with your own data. I removed all  sample data that I inputed initially. You can test with yours . 

Regards

tresorsawasawa commented:
@ogagaoghene I don't have your review anymore. It was already claimed by someone else. But it will be difficult to test your app without fixing that error. 

But if you have fixed that, you will get review soon

Davidosky007 commented on README.md:
```diff
@@ -1,54 +1,235 @@
+### Run tests
+
+To run tests, run the following command:
+- bundle exec rspec spec/feature/test_spec.rb
```
* Kindly fix these errors i get when i try to run the tests
![image](https://user-images.githubusercontent.com/52497246/203882522-35bd3d0a-f07c-41f4-bb9c-7cba053454d8.png)

ogagaoghene replied:
@Davidosky007 Am I supposed write tests. I did not write any. Those are default test cases. Writing tests is not part of the requirement. Kindly clarify on this please

ogagaoghene replied:
@Davidosky007 I did intend to put that line. But, in the new README file which I saw a line. I just followed through by giving an example of how to write tests. If you check the requirements, writing of tests is not part of it. Could you please revert on this?

ogagaoghene replied:
@Davidosky007 Thanks for your professional code review. I am the only one doing the work of 2 persons. I am exhausted but what can I do. I have fixed the errors with the tests you requested, You can try again by running the command below:
-
bundle exec rspec spec/tests/

Davidosky007 commented on README.md:
```diff
@@ -1,54 +1,235 @@
 
+### Setup
 
-### Run project
```
* I strongly suggest you properly update your readme file setup section on how to properly install and run your app by any visitor of  your repo ( which might be a non tech person ) its not enough for your project to work locally at your end alone, take the visitor step by step on how to achieve setup success as seen at your end 🙏 
* i managed to install it on my end by fixing a lot of migration issues 😓  which might not be the case for a non tech visitor 

> By running `rails db:drop:all` `rails db:create` `rails db:migrate`

> And i removed all duplicate columns ie ( confirmation_token ) which is already existing 
![image](https://user-images.githubusercontent.com/52497246/203884684-ea0a337f-21eb-4eb8-8168-431524a175b2.png)
![image](https://user-images.githubusercontent.com/52497246/203884757-71621e4a-5661-4884-bd2f-0239fcd95d75.png)


![image](https://user-images.githubusercontent.com/52497246/203883695-837a7a50-20bd-4071-9a05-c19029d29120.png)
![image](https://user-images.githubusercontent.com/52497246/203883854-822d3774-0697-4669-bbae-b12b59d4ccde.png)

ogagaoghene replied:
@Davidosky007 I have updated the README file as requested, I am exhausted bro. I anticipate a positive feedback so I can attend to other pressing issues. Thanks in anticipation.

Davidosky007 reviewed (changes requested):
Hi @ogagaoghene 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 
* Descriptive commit messages 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Munsa1 reviewed (changes requested):
Hi @ogagaoghene, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is not runnig

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

ogagaoghene commented:
@Munsa1 I don't understand you bro.  Please find screenshots of what you are saying is not running. A little bit of patience is needed bro. I am tired of this frustration. Where is the professionalism in this practice? Maybe I need to report this issue to Microverse support because it is getting out of hand. A colleage of yours was able to run it successfully and asked that I should fix the errors from the tests and add it to project's README file. I have done that and you are invalidating my request. What is happening here?

The commands he said he issued to get it working at his end are as outlined: 
rails db:migrate:drop:all
rails db:migrate:create
rails db:migrate

I have included all of these in project's README file. What else do you guys want from me? I am running out of patience. I need to make a case for this situation outside the scope of code reviewer's job because it is frustrating. 

![image](https://user-images.githubusercontent.com/30408871/203985905-2cb06713-83c1-41d8-a98b-918df00d88dc.png)
![image](https://user-images.githubusercontent.com/30408871/203986062-889704c6-7cc3-4a48-8b62-45f94be25083.png)
![image](https://user-images.githubusercontent.com/30408871/203986125-a33b9824-5f44-4f33-ab44-9b831de0047d.png)
![image](https://user-images.githubusercontent.com/30408871/203986157-0398b163-0a54-4ee2-839d-a71727313e72.png)

ogagaoghene commented:
@Munsa1 What your professional colleagues normally do is to initiate a zoom call to verify my claim and to see how to assist the student. But, I can't sense professionalism. Code review is not supposed to be a malicious act. Where is the mentorship and professionalism here?

The first instance it was the issue of a README file
The second instance was one that has to do with a colleague of yours that could not get it to run locally
The third instance, a colleague of yours was able to get it to run locally and requested that I should just fix the errors from the test
The fourth instance which is your situation is outright invalidation of my request. 

This is so sad and painful I must say. 

What am I doing here if I am surrounded by unfriendliness and unprofessionalism?

Davidosky007 commented:
> @Munsa1 What your professional colleagues normally do is to initiate a zoom call to verify my claim and to see how to assist the student. But, I can't sense professionalism. Code review is not supposed to be a malicious act. Where is the mentorship and professionalism here?
> 
> 
> 
> The first instance it was the issue of a README file
> 
> The second instance was one that has to do with a colleague of yours that could not get it run locally
> 
> The third instance, a colleague of yours was able to get it to run locally and requested that I should just fix the errors from the test
> 
> The fourth instance which is your situation is outright invalidation of my request. 
> 
> 
> 
> This is so sad and painful I must say. 
> 
> 
> 
> What am I doing here if I am surrounded by unfriendliness and unprofessionalism? 
> 
> 

Are you free for a call @ogagaoghene 
I will suggest you request a second opinion

ogagaoghene commented:
@Munsa1 I can rememer a time Ajise Toluwase, a fellow code reviewer spent quite some time ensuring that a situation was professionally handled. That's the spirit of coding in a professional setting. 

I sincerely need assistance in any form because this back and forth is killing my spirit especially when I have put in a whole lot to get this to work completely.

andersonlebon reviewed (approved):
### Hello @ogagaoghene 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

![nodding-yes](https://user-images.githubusercontent.com/65068771/143095390-344d1cb0-b853-4b91-8662-5d9049dc9e60.gif)

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------

ogagaoghene commented:
@andersonlebon Many thanks you bro. I am eternally. I feel better, Thank you so very much bro. May God bless you real good.
